cems-api.yml

```diff
@@ -7,1054 +7,6172 @@ host: "api.hceu-sandbox.com"
+                  properties: 
+                    name: 
+                      type: "string"
+                      example: "Add/Edit Users"
+                    constant: 
+                      type: "string"
+                      example: "add-edit-users"
+                    description: 
+                      type: "string"
+                      example: "Allow university corp admins to add/edit users"
+        401:
+          description: "Not authenticated"
+        403:
+          description: "Not authorized"
+        404:
+          description: "Company or resource not found"
+          
+  /company/{company_id}/corpadminreport:
+    get:
+      tags:
+        - "Reports"
+      summary: "Get all corp admin reports for a company"
+      description: ""
+      parameters:
+        - in: "path"
+          name: "company_id"
+          description: ""
+          required: true
+          type: "integer"
+          pattern: '\d+'
+      responses:
+        200:
+          description: "Success"
+          schema:
+            $ref: "#/definitions/CorpAdminReport"
+        401:
+          description: "Not authenticated"
+        403:
+          description: "Not authorized"
+        404:
+          description: "Company not found"
+          
+  /company/{company_id}/course/self-enroll:
+    post:
+      tags:
+        - "User"
+      summary: "Registers the currently logged in user into a given course"
+      description: ""
+      parameters:
+        - in: "path"
+          name: "company_id"
+          description: ""
+          required: true
+          type: "integer"
+          pattern: '\d+'
+        - name: "body"
+          in: "body"
+          description: ""
+          required: true
+          schema:
+            type: "object"
+            required: 
+              - "course_id"
+            properties: 
+              course_id: 
+                type: "string"
+                example: "9999"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            $ref: "#/definitions/SelfEnroll"
+        401:
+          description: "Not authenticated"
         403:
-          description: "user was authenticated but not authorized"
+          description: "Not authorized"
 
-definitions:
-  Enrollment:
-    type: "object"
+  /company/{company_id}/course_type:
+    get:
+      tags:
+        - "Course"
+      summary: "Get list of course types"
+      description: ""
+      parameters:
+        - in: "path"
+          name: "company_id"
+          description: ""
+          required: true
+          type: "integer"
+          pattern: '\d+'
+      responses:
+        200:
+          description: "Success"
+          schema:
+            $ref: "#/definitions/CourseType"
+        401:
+          description: "Not authenticated"
+        403:
+          description: "Not authorized"
+        404:
+          description: "Company not found"
+          
+  /company/{company_id}/courses:
```

Comment:
Sorry Cory, I had already written docs for this endpoint but removed them because Genesis couldn't use it... here is a much more detailed version:

```yaml
paths:
  /company/{companyId}/courses:
    get:
      tags:
        - "Company"
      summary: "Company Course Catalog"
      description: "search for courses within a company, warning: this endpoint auto applies filters based on what the requesting user is elegable to take, do not use to get all courses"
      operationId: "getCompanyCourses"
      parameters:
        - name: "companyId"
          in: "path"
          description: ""
          required: true
          type: "integer"
        - name: "filter[keyword][value]"
          in: "query"
          description: "filter by course name"
          required: false
          type: "string"
        - name: "filter[hours][value]"
          in: "query"
          description: "filter by credit hours"
          required: false
          type: "integer"
        - name: "filter[state][value]"
          in: "query"
          description: "ex: TX"
          required: false
          type: "string"
        - name: "filter[item_format][value]"
          in: "query"
          description: "ex: Video"
          required: false
          type: "string"
        - name: "page"
          in: "query"
          description: "used with limit to break list into pages"
          required: false
          type: "integer"
        - name: "limit"
          in: "query"
          description: "limit the number of results"
          required: false
          type: "integer"
        - name: "sort"
          in: "query"
          description: "sort by field"
          required: false
          type: "string"
        - name: "direction"
          in: "query"
          description: "sort direction: asc | desc"
          required: false
          type: "string"
      responses:
        200:
          description: "success"
          schema:
            $ref: "#/definitions/CourseCatalog"
        403:
          description: "user was authenticated but not authorized"
        404:
          description: "Company not found"

 
definitions:
  CourseCatalog:
    required: 
      - "data"
    properties: 
      data:
        type: "object"
        required: 
          - "current_page"
          - "items_per_page"
          - "total"
          - "items"
        properties: 
          current_page: 
            type: "number"
            example: 1
          items_per_page: 
            type: "number"
            example: 10
          total: 
            type: "number"
            example: 114248
          items: 
            type: "array"
            items: 
              type: "object"
              properties: 
                id: 
                  type: "number"
                  example: 7191
                name: 
                  type: "string"
                  example: "Course Title"
                description: 
                  type: "string"
                  example: "course description"
                hours: 
                  type: "number"
                  example: 1.5
                category_name: 
                  type: "string"
                  example: "Compliance Training"
                approval_status_name: 
                  type: "string"
                  example: "Professional Development - No CE Credit"
                item_format_name: 
                  type: "string"
                  example: "Video"
                preview_link: 
                  type: "string"
                preview_name: 
                  type: "string"
                is_credit: 
                  type: "boolean"
                  example: true
                incomplete: 
                  type: "string"
                  example: "0"
                is_soft_due_date: 
                  type: "string"
                  example: "hard"
```

you will notice the description ... and the filter query arguments.  it is important to document the query arguments also

Reply:
I've been including filter query arguments where I see them but the way this endpoint is documented above is completely different from the current documentation and it returns a 200 without any of those filters so I just had no idea they needed to be included. I can update my documentation to reflect what you have above though.